src/main/scala/Main.scala

```diff
@@ -53,14 +53,25 @@ class SharedAxisCharts extends VBox {
 
     val lastChart = charts.last.chart
 
-    if (event.isShiftDown) {  // TODO implement vertical zoom/pan
+    val (newYLower, newYUpper) = if(event.isShiftDown && event.isControlDown){ //vertical scroll
```

Comment:
I think you have the modifiers mixed up? ctrl+shift should zoom instead of scroll - the idea being that ctrl is the global zoom modifier, while shift is the vertical modifier

Also, style: space between `if` and opening parens, and between close parens and opening bracket